### PR TITLE
refactor(ui5-segmented-button): rename `mode` to `selectionMode`

### PR DESCRIPTION
**Pull request by tsanislavgatev**
Change the enum name from `SegmentedButtonMode` to `SegmentedButtonSelectionMode`.
The options in the enum are changed from `SingleSelect` and `MultiSelect` to `Single` and `Multiple`.
The `mode` property of `SegmentedButton` is changed to `selectionMode`.
Also deleted deprecated getter - `selectedItem` and deprecated event detail `selectedItem`.

BREAKING CHANGE: The mode property is changed to `SelectionMode` and the values it take from `SingleSelect` and `MultiSelect` to `Single` and `Multiple`. Also deleted deprecated getter - `selectedItem` and deprecated event detail `selectedItem` and now can be used with the `selectedItems` getter and `selectedItems` event detail.

Previously the application developers could set the selection mode as follows:
```ts
<ui5-segmented-button id="segButtonMulti" mode="MultiSelect">
  <ui5-segmented-button-item>Item</ui5-segmented-button-item>
  <ui5-segmented-button-item>Item</ui5-segmented-button-item>
  <ui5-segmented-button-item>Click</ui5-segmented-button-item>
  <ui5-segmented-button-item>SegmentedButtonItem</ui5-segmented-button-item>
</ui5-segmented-button>
```
Now the application developers could set the selection mode as follows:
```ts
<ui5-segmented-button id="segButtonMulti" selection-mode="Multiple">
  <ui5-segmented-button-item>Item</ui5-segmented-button-item>
  <ui5-segmented-button-item>Item</ui5-segmented-button-item>
  <ui5-segmented-button-item>Click</ui5-segmented-button-item>
  <ui5-segmented-button-item>SegmentedButtonItem</ui5-segmented-button-item>
</ui5-segmented-button>
```
Related to: https://github.com/SAP/ui5-webcomponents/issues/8461

### CODE REVIEW

File: packages/main/src/types/SegmentedButtonSelectionMode.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 	 */
-	SingleSelect = "SingleSelect",
+	Single = "Single",
 
 	/**
 	 * Multiple items can be selected at a time. All items can be deselected.
 	 * @public
 	 */
-	MultiSelect = "MultiSelect",
+	Multi = "Multi",
```
Comment (unazko) [inline packages/main/src/types/SegmentedButtonSelectionMode.ts:16]: Shouldn't this be "Multiple" as in the `ui5-calendar` component?

**Comment by unazko** [comment]
The breaking change would be about the changing the name of the `selection `property to `selectionMode` and the type it takes.
It has to be adjusted in the PR message.

File: packages/main/src/types/SegmentedButtonSelectionMode.ts (hunk 1 above)
Comment (tsanislavgatev) [reply packages/main/src/types/SegmentedButtonSelectionMode.ts:16]: Yes, it should.

**Review by Todor-ads — APPROVED** [review]

File: packages/main/src/SegmentedButton.ts [hunk 2]
```diff
@@ -71,11 +70,6 @@ type SegmentedButtonSelectionChangeEventDetail = {
  */
```
Comment (nnaydenow) [inline packages/main/src/SegmentedButton.ts:70]: jsdoc tags need to be aligned too

**Review by nnaydenow — CHANGES_REQUESTED** [review]

**Review by nnaydenow — APPROVED** [review]

**Comment by ilhan007** [comment]
Hi @tsanislavgatev 
shouldn't you also update the oficial samples, the MultiSelect sample is currently broken after this change:
https://sap.github.io/ui5-webcomponents/nightly/components/SegmentedButton/#selection-modes

**Comment by ilhan007** [comment]
The issue will be fixed in https://github.com/SAP/ui5-webcomponents/pull/8869/files

**Pull request merged**